mountpoint-s3-fs/src/superblock.rs

```diff
@@ -193,11 +196,11 @@ impl<'a> RenameLockGuard<'a> {
         // Take read lock
         let ancestor_check = Self::dst_is_ancestor(&superblock_inner.inodes.read().unwrap(), source_parent, dst_ino);
         if ancestor_check || src_ino > dst_ino {
-            dst_parent_lock = Some(dest_parent.get_mut_inode_state()?);
-            src_parent_lock = source_parent.get_mut_inode_state()?;
+            dst_parent_lock = Some(dest_parent.get_mut_inode_state()?.into());
```

Comment:
minor: not too convinced this `.into()` is worth it. It's not transparent and even longer than `.state`. Also not used consistently, e.g. line 189.

 Why not just use `InodeLockedForWriting` in `RenameLockGuard`?